docs/source/playbook.rst

```diff
@@ -0,0 +1,191 @@
+Writing actions
+===============
+
+All Ansible is contained in `obal/data`. There we have `playbooks` and `roles`.
+
+A `playbook` with `metadata` is considered an action and exposed to the user as such.
+
+Writing playbooks
+-----------------
+
+We have a slightly non-standard playbooks layout. Every playbook is contained in its own directory and named `playbook.yaml`. It can also contain a `metadata.yaml`. While playbooks are pure ansible, the metadata is the data Obal needs to extract to build a CLI.
+
+Obal uses "hosts" to operate on. If you have a set of packages then every package is a host. There are some special hosts:
+
+* localhost
+* packages
+
+As with regular Ansible, `localhost` is used to operate on the local machine. This is typically used to setup or work on environments.
+
+Packages is the entire set of all packages. These are exposed on the command line to users so they can operate on a limited set of packages.
+
+Note that hosts must always be a list. Our setup playbook is an example of a local connection:
+
+.. literalinclude:: ../../obal/data/playbooks/setup/playbook.yaml
+  :language: yaml
+  :caption: setup.yaml
+  :emphasize-lines: 2,3
+
+When dealing with packages we typically include the `package_variables` role to set various variables:
+
+.. literalinclude:: ../../obal/data/playbooks/changelog/playbook.yaml
+  :language: yaml
+  :caption: playbook.yaml
+  :emphasize-lines: 6,7
+
+Exposing playbooks using metadata
+---------------------------------
+
+By default Obal exposes a playbook based on its name. It can also automatically detect whether it accepts a packages parameter. To provide a better experience we introduce metadata via `metadata.yaml` in the same directory.
```

Comment:
I think we could change `we introduce` to `better experience there is metadata provided via...`